timeseries/src/autogluon/timeseries/models/chronos/pipeline/chronos_bolt.py

```diff
@@ -223,20 +325,108 @@ def _init_weights(self, module):
             module.output_layer.weight.data.normal_(mean=0.0, std=factor * ((self.config.d_ff) ** -0.5))
             if hasattr(module.output_layer, "bias") and module.output_layer.bias is not None:
                 module.output_layer.bias.data.zero_()
+        elif isinstance(module, SelectionBlock):
+            d_model = self.config.d_model
+            key_value_proj_dim = self.config.d_kv
+            n_heads = self.config.num_heads
+            module.q.weight.data.normal_(mean=0.0, std=factor * ((d_model * key_value_proj_dim) ** -0.5))
+            module.k.weight.data.normal_(mean=0.0, std=factor * (d_model**-0.5))
+            module.v.weight.data.normal_(mean=0.0, std=factor * (d_model**-0.5))
+            module.o.weight.data.normal_(mean=0.0, std=factor * ((n_heads * key_value_proj_dim) ** -0.5))
+
+    def _past_feat_patch_and_embed(self, tensor: torch.Tensor) -> torch.Tensor:
```

Comment:
I haven't checked this part of the code since I don't have a strong intuition about it yet. I will take a closer look once we have some reasonable benchmark numbers